src/components/Section.js

```diff
@@ -147,7 +147,7 @@ export default class Section extends React.Component {
             onPostVote={this.onPostVote.bind(this)}
             comments={comments}
             pluginPurpose="postComments"
-            canSetNickname={user === null}
+            defaultNickname={user && user.displayName}
```

Comment:
Here, the user default nickname should actually be obtained from the backend. I'm adding a `nickname` field to the API users endpoint.

Therefore, here we should change the `/me` endpoint to contain an additional user field `nickname` that is populated from the API `nickname` field, if the API has such a field. Then, we could set the nickname to be the one stored in the API.

The `user.displayName` should remain the user's real name, visible only to the user himself but not posted as nickname if a nickname already exists.